e2e/test_operator/operator_test.go

```diff
@@ -142,6 +144,66 @@ var _ = Describe("Operator", Label("e2e"), func() {
 		})
 	})
 
+	When("taint a Pod", func() {
+		var replacedPod corev1.Pod
+		taintKeyMaintenance := "maintenance"
+		taintKeyMaintenanceDuration := int64(2)
+		taintKeyStar := "*"
+		taintKeyStarDuration := int64(5)
+
+		BeforeEach(func() {
+			initialPods := fdbCluster.GetStatelessPods()
+			replacedPod = fixtures.RandomPickOnePod(initialPods.Items)
+			replacedPodProcessGroupID := internal.GetProcessGroupIDFromMeta(fdbCluster.GetCluster(), replacedPod.ObjectMeta)
+			// Taint replacePod's node
+			node := &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{Name: replacedPod.Spec.NodeName},
+			}
+			node.Spec.Taints = []corev1.Taint{
+				{
+					Key:       taintKeyMaintenance,
+					Value:     "rack_maintenance",
+					Effect:    corev1.TaintEffectNoExecute,
+					TimeAdded: &metav1.Time{Time: time.Now().Add(-time.Second * time.Duration(taintKeyMaintenanceDuration+1))},
+				},
+			}
+			log.Printf("Taint node: Node name:%s Node taints:%+v TaintTime:%+v Now:%+v\n", replacedPod.Name,
+				node.Spec.Taints, node.Spec.Taints[0].TimeAdded.Time, time.Now())
+			err := fdbCluster.UpdateNode(node)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Setup cluster's taint config and wait for long enough
+			err = fdbCluster.SetClusterTaintConfig([]fdbv1beta2.TaintReplacementOption{
+				{
+					Key:               &taintKeyStar,
+					DurationInSeconds: &taintKeyStarDuration,
+				},
+				{
+					Key:               &taintKeyMaintenance,
+					DurationInSeconds: &taintKeyMaintenanceDuration,
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(fdbCluster.GetAutomationOptions().Replacements.Enabled).To(BeTrue())
+			processGroupStatus := fdbv1beta2.FindProcessGroupByID(fdbCluster.GetCluster().Status.ProcessGroups, replacedPodProcessGroupID)
+			Expect(processGroupStatus).NotTo(BeNil())
+			Expect(processGroupStatus.GetCondition(fdbv1beta2.NodeTaintDetected)).NotTo(BeNil())
+			Expect(processGroupStatus.GetCondition(fdbv1beta2.NodeTaintReplacing)).NotTo(BeNil())
+
+			// Wait for operator to replace the pod
+			time.Sleep(time.Second * time.Duration(*fdbCluster.GetAutomationOptions().Replacements.FailureDetectionTimeSeconds))
```

Comment:
What is the benefit of the sleep?

Reply:
Cluster won't replace the pod until the pod has been in the TaintReplacing condition for Replacements.FailureDetectionTimeSeconds